src/codegen/TypeScriptEngine.ts

```diff
@@ -30,7 +30,20 @@ export default class TypeScriptEngine {
             tsParams,
             tsReturnType,
             undefined
-        )
+        );
+
+        const comments: string[] = funDecl.getComments();
+        if (comments.length) {
+            const commentString: string = this.generateCommentString(comments);
+            ts.addSyntheticLeadingComment(
+                funcDeclaration,
+                SyntaxKind.MultiLineCommentTrivia,
+                commentString,
+                true
+            );
+        }
```

Comment:
Do you mind if we extract this logic out into a private helper? We could then reuse this in other AST generation functions, e.g. `createInterface`